nixos/modules/services/databases/mysql.nix

```diff
@@ -21,6 +21,8 @@ let
   installOptions =
     "${mysqldOptions} ${lib.optionalString isMysqlAtLeast57 "--insecure"}";
 
+  configFile = pkgs.writeText "my.cnf" (generators.toINI {} cfg.config + optionalString (cfg.extraOptions != null) "[mysqld]\n${cfg.extraOptions}");
```

Comment:
I guess merging multiple `mysqld` sections works in this case?

Reply:
Yes, sections headers can be repeated any number of times.